### ISSUE TITLE
Channel order

### ISSUE THREAD

**Issue by wongdanr** [broadinstitute/lincs-cell-painting#85]
Hello! Where can I find which markers/stains each of the 5 image channels correspond to?

**Comment by gwaygenomics** [comment]
Hi @wongdanr - I'll point you to a couple resources:

- Channel/Structure map ([link](https://github.com/broadinstitute/lincs-profiling-complementarity/blob/b5898bfce59b0fd8bd5e02b7bd5cafe743312c66/6.paper_figures/example_images/utils/channels.py#L8))
- Code to map image channel to marker ([link](https://github.com/broadinstitute/lincs-profiling-complementarity/blob/b5898bfce59b0fd8bd5e02b7bd5cafe743312c66/6.paper_figures/example_images/utils/parse_xml.py#L101))
    - [How to download XML files](https://github.com/broadinstitute/lincs-profiling-complementarity/blob/b5898bfce59b0fd8bd5e02b7bd5cafe743312c66/6.paper_figures/example_images/0.download_xml.sh)

To summarize:

| Channel | Stain | Structure | ID |
| :-------: | :---: | :--------: | :--: |
| 1 | HOECHST 33342 | DNA | `ch1` |
| 2 | Alexa 488 | ER | `ch2` |
| 3 | 488 long | RNA | `ch3` |
| 4 | Alexa 568  | AGP | `ch4` |
| 5 | Alexa 647  | Mito | `ch5` |

**Comment by bethac07** [comment]
Yes, the xml files are the source of truth

**Comment by wongdanr** [comment]
Sorry @bethac07 so does this channel mapping not apply to all LINCS images? 
ch01 - HOECHST 33342 DNA
ch02 - Alexa 488 ER 
ch03 - 488 long RNA
ch04 - Alexa 568 AGP
ch05 - Alexa 647 Mito

**Comment by bethac07** [comment]
That mapping may or may not, I don't know. I have no evidence that it does not, but I do know we have occasionally had cases where within batches (and certainly across them) the channel order changes between plates. It doesn't sound like anyone in anything documented has ever confirmed if this is consistent.

**Comment by bethac07** [comment]
The xml files have the mappings, but as I stated, I would not trust them at a batch level, only at a plate level.

**Comment by wongdanr** [comment]
Thanks @bethac07, I just parsed through all of the xml files and confirmed that indeed the mapping that @gwaygenomics gave holds for all plates (at least the ones in s3://cellpainting-gallery/lincs/broad/images/2016_04_01_a549_48hr_batch1/images/ xml/). Thank you!

**Comment by gwaygenomics** [comment]
Wonderful! Thanks for following up with this @wongdanr

If possible, can you paste the code snippet you used to confirm? (It is likely to help future users who stumble upon this issue)

**Comment by wongdanr** [comment]
sure thing @gwaybio: 

`def testChannelOrderThroughLINCSXMLFiles():
    """
    Make sure that our lincs channel image assumption is correct: 
    {1: "HOECHST 33342", 2:"Alexa 488", 3:"488 long", 4:"Alexa 568", 5:"Alexa 647"}
    Parses through xml files in /home/wongd26/workspace/profiler/lincs_xml_files/xml/
    (aws s3 cp --recursive --no-sign-request s3://cellpainting-gallery/lincs/broad/images/2016_04_01_a549_48hr_batch1/images/ xml/ --exclude "*" --include "*.xml")
    """
    directory = "/home/wongd26/workspace/profiler/lincs_xml_files/xml/"
    channel_map = {1: "HOECHST 33342", 2:"Alexa 488", 3:"488 long", 4:"Alexa 568", 5:"Alexa 647"}
    for sub_dir in os.listdir(directory):
        print("subdir", sub_dir)
        with open(directory + sub_dir + "/Images/Index.idx.xml", "r") as xmlfile:
            lines = xmlfile.readlines()
            for line in lines:
                if "<ChannelID>" in line:
                    channel = int(line[line.find("<ChannelID>") + 11: line.rfind("<")])
                if "<ChannelName>" in line:
                    name = line[line.find("<ChannelName>") + 13: line.rfind("<")]
                    assert(channel_map[channel] == name)`

**Comment by gwaybio** [comment]
awesome. thanks @wongdanr
